src/charm.py

```diff
@@ -149,8 +149,11 @@ def _on_remove(self, event: RemoveEvent):
         if root_token:
             vault = Vault(url=self._api_address)
             vault.set_token(token=root_token)
-            if vault.is_api_available() and vault.node_in_raft_peers(node_id=self._node_id):
-                vault.remove_raft_node(node_id=self._node_id)
+            try:
+                if vault.is_api_available() and vault.node_in_raft_peers(node_id=self._node_id):
+                    vault.remove_raft_node(node_id=self._node_id)
+            except (requests.exceptions.ConnectionError, requests.exceptions.TooManyRedirects):
+                pass
```

Comment:
I am curious when those exceptions could raise. I would guess when it is the last unit?

Would it make sense to log something here?

Reply:
I found why we were getting the `TooManyRedirectErrors` (the bind_address) is not always available on remove event. This  try/except is not necessary anymore and it's been removed